File: feature_1_async/example1.dart
Future<String> getHello() async {
  print('waiting for data');
  // Das könnte auch ein HTTP Request sein
  final response = await Future.delayed(Duration(seconds: 2), () {
    print('data received');
    return 'Hello World!';
  });
  return response;
}

void main(List<String> args) async {
  final value = await getHello();
  print('got value $value');
}

File: feature_1_async/Example1.java
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.TimeUnit;

public class Example1 {
  public static CompletableFuture<String> getHello() {
    System.out.println("waiting for data");

    // Das könnte auch ein HTTP Request sein
    return CompletableFuture.supplyAsync(() -> {
      try {
        TimeUnit.SECONDS.sleep(2);
        System.out.println("data received");
      } catch (InterruptedException e) {
        e.printStackTrace();
      }
      return "Hello World!";
    });
  }

  public static void main(String[] args) {
    CompletableFuture<String> future = getHello();

    try {
      var value = future.get();
      System.out.println("got value: " + value);
    } catch (InterruptedException | ExecutionException e) {
      e.printStackTrace();
    }
  }

}
